### PR TITLE
Metamask wallet connection and sign-in flow

### PR DESCRIPTION
**Pull request by AliBhutta2073**

### CODE REVIEW

File: backend/src/socket/ndauConnect.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ module.exports = (_io) => {
     console.log(socket.id);
```
Comment (moneta) [inline backend/src/socket/ndauConnect.js:26]: Please just rename the file name ndauConnect to, says, walletConnect or something similar

File: frontend/src/Components/buttons/connect_metamask_button.jsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        if(metamaskWeb3.walletAddress){
+           
+            socketEmit("ndau_burn_wallet_connect", {
+                website_socket_id: socket.id,
+                app_socket_id: "31231",
+                action: "burn",
+                wallet_address:metamaskWeb3.walletAddress,
+            });
+
+            socket.on(
```
Comment (moneta) [inline frontend/src/Components/buttons/connect_metamask_button.jsx]: We should get rid of these sockets

File: frontend/src/Components/VoteButton/VoteButton.tsx [hunk 3]
```diff
@@ -104,44 +107,69 @@ const VoteButton = ({ dynamicClassName, allowVote, selectedVoteOption }: VoteBut
...
+    _payload = btoa(yaml.stringify(_payload));
 
-  const signManually = async ({ signature }: FormInputsI) => {
-    console.log('payload, signature', payload, signature);
     try {
       const resp = await axiosRequest('post', 'vote', {
-        payload,
-        signature,
+        payload: _payload,
+        signature
```
Comment (moneta) [inline frontend/src/Components/VoteButton/VoteButton.tsx]: This should be the signature against the complete payload, not the sign-in signature. The signature should use  typed struct V4

**Review by moneta — CHANGES_REQUESTED** [review]

**Comment by AliBhutta2073** [comment]
Closing, will create a new PR with changes.

**Pull request closed without merging**